### PR TITLE
add subcompaction listener

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

Add listener for subcompaction, only called when there are actually multiple subcompactions executed in parallel.
Pass in `SubcompactionJobInfo` containing:
```
  // the id of the column family where the compaction happened.
  uint32_t cf_id;
  // the name of the column family where the compaction happened.
  std::string cf_name;
  // the status indicating whether the compaction was successful or not.
  Status status;
  // the id of the thread that completed this compaction job.
  uint64_t thread_id;
  // the smallest input level of the compaction.
  int base_input_level;
  // the output level of the compaction.
  int output_level;
```

### CODE REVIEW

**Review by Connor1996 — COMMENTED** [review]
what's it used for?

**Comment by tabokie** [comment]
> what's it used for?

@Connor1996 Check this comment: https://github.com/tikv/rocksdb/pull/216#discussion_r539068471, used for tagging subcompaction IO type.

**Review by yiwu-arbug — APPROVED** [review]
LGTM

File: include/rocksdb/listener.h [hunk 1]
```diff
@@ -233,6 +233,23 @@ struct CompactionJobInfo {
...
+
+  // the id of the column family where the compaction happened.
+  uint32_t cf_id;
+  // the name of the column family where the compaction happened.
+  std::string cf_name;
+  // the status indicating whether the compaction was successful or not.
+  Status status;
+  // the id of the thread that completed this compaction job.
+  uint64_t thread_id;
+  // the smallest input level of the compaction.
```
Comment (Connor1996) [inline include/rocksdb/listener.h:247]: Add job_id so we can use it to match the CompactionJobInfo later?

Comment (tabokie) [reply include/rocksdb/listener.h:247]: It seems job id is only unique within each background thread, that's why I removed it.
```c++
// the job id, which is unique in the same thread.
  int job_id;
```

Comment (tabokie) [reply include/rocksdb/listener.h:247]: We can put more thoughts on how to match subcompaction with main compaction later. Right now this PR covers our user case.

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**